front/src/applications/stdcm/utils/formatSimulationReportSheet.ts

```diff
@@ -106,6 +130,121 @@ export function getStopDurationBetweenTwoPositions(
   return null;
 }
 
+/**
+ * @param op Operational point to format
+ * @param train Object containing simulated train positions, times, and departure time
+ * @param simulationPathSteps List of simulation path steps
+ * @returns A formatted operational point with calculated stop duration and departure time
+ */
+function formatOperationalPointWithTimes(
+  op: SuggestedOP,
+  train: TrainSimulationProps,
+  simulationPathSteps: StdcmPathStep[]
+): StdcmResultsOperationalPoint {
+  const formattedTime = getTimeAtPosition(op.positionOnPath, train);
+
+  const duration = getStopDurationAtPosition(op.positionOnPath, train.positions, train.times);
+  const durationInSeconds = duration !== null ? duration / 1000 : 0;
+  const durationToString = secondsToTimeString(durationInSeconds);
+  const stopEndTime = computeStopDepartureTime(formattedTime, durationToString);
+  // Find the corresponding stopType from pathSteps
+  const correspondingStep = simulationPathSteps.find(
+    (step) => step.location && matchPathStepAndOp(step.location, op)
+  );
+  let stopType;
+  if (correspondingStep) {
+    stopType = !correspondingStep.isVia ? 'serviceStop' : correspondingStep.stopType;
+  }
+  const stopFor = correspondingStep?.isVia ? correspondingStep.stopFor : undefined;
+
+  return {
+    opId: op.opId!,
+    positionOnPath: op.positionOnPath,
+    time: formattedTime,
+    name: op.name,
+    ch: op.ch,
+    duration: durationInSeconds,
+    stopEndTime,
+    trackName: op.metadata?.trackName,
+    stopType,
+    stopFor,
+  };
+}
+
+/**
+ * @param finalOutput Final simulation report containing lists of positions and times of all simulated points
+ * @returns A list of all positions at which the trains stops
+ */
+function findAllStops(finalOutput: ReportTrain): number[] {
+  return finalOutput.positions.filter(
+    (position, index) =>
+      index !== finalOutput.positions.length &&
```

Comment:
The array length check needs to be fixed to properly handle the last position:
```suggestion
      index !== finalOutput.positions.length -1 &&
```

Reply:
Good catch my bad!